lib/check-dependencies.js

```diff
@@ -104,6 +104,9 @@ const checkDependenciesHelper = (syncOrAsync, config, callback) => {
         return finish();
     };
 
+    const supportsPackageAliases = packageManager =>
+        packageManager === 'npm' || packageManager === 'yarn';
```

Comment:
Do we need this check? I imagine some other package managers like pnpm may also support aliases. We might want to exclude Bower here as that's completely different but in the rest I wouldn't assume lack of support.

Reply:
@mgol Yes, keeping a whitelist is secure, keeping a blacklist will eventually cause issues. All it takes is one package manager that interprets the colon differently to have huge discrepancies rendering the package unusable. Either way, when support for these other package managers is added code like this can be changed.

Reply:
By the way, not having sufficient checks like this is what prompted this PR in the first place, you take aliased version structure and pretend to know what it is, causing incorrect behaviour. Care should be taken to keep publicly used code forwards compatible